src/components/forms/CampaignDetailsForm/useSelectManager.tsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+
+import getPeopleSearchResults from 'fetching/getPeopleSearchResults';
+import useDebounce from 'hooks/useDebounce';
+import { useQuery } from 'react-query';
+import { useRouter } from 'next/router';
+import { useEffect, useState } from 'react';
+
+import { useUser } from 'hooks';
+
+const useSelectManager = (initialManager?: {id: number; name: string} | null) => {
+    const { orgId } = useRouter().query;
+
+    // People Search Results
+    const [managerSearchValue, setManagerSearchValue] = useState<string>(initialManager?.name || '');
+    const managerSearchQuery = useQuery(
+        ['peopleSearchResults', managerSearchValue],
+        getPeopleSearchResults(managerSearchValue, orgId as string),
+        { enabled: false },
+    );
+
+    // Handle search when managerSearchValue changes
+    const debouncedQuery = useDebounce(async () => {
+        managerSearchQuery.refetch();
+    }, 600);
+    useEffect(() => {
+        if (managerSearchValue.length >= 3) {
+            debouncedQuery();
+        }
+    }, [managerSearchValue.length, debouncedQuery]);
+
+    // Setting the selected Manager
+    const [selectedManager, setSelectedManager] = useState(initialManager);
+    const user = useUser();
+
+    return {
+        managerSearchQuery,
+        managerSearchValue,
+        selectedManager,
+        setManagerSearchValue,
+        setSelectedManager,
+        setSelectedManagerToUser: () => user && setSelectedManager({ id: user.id, name: `${user.first_name} ${user.last_name}` }),
```

Comment:
If I'm not mistaken, there is a bug here, that did not exist in my original code.

This line of code is setting the manager to the `user` object, but the manager is not a _user_, it's a _person_. The `Person` object in Zetkin is not the same as `User`, and setting the `manager_id` to `user.id` (which is the end result of this) will result in another person (with the ID of the user's `User` object) being added as the manager.

This is what the `getUserMemberships()` function was for. It retrieves the memberships of the current user, in which the `Person` object for this specific organization can be found. It seems like this function is no longer being used here.

The reason why this issue does not become apparent with the dummy data, is that the ID of the `testadmin@example.com` user (2) is the same as the `Person` it's connected to in organization 1.